### PR TITLE
bugfix: fixed the bugs in child contracts calls

### PR DESCRIPTION
**Pull request by d5c5ceb0**
1  load old data from db;
2  get old balance;
3  logs is not stored;

### CODE REVIEW

File: sealEVM.go [hunk 1]
```diff
@@ -75,6 +75,25 @@ func New(param EVMParam) *EVM {
...
+		context:      param.Context,
+		instructions: nil,
+		resultNotify: param.ResultCallback,
+	}
+
+	evm.instructions = instructions.New(evm, evm.stack, evm.memory, evm.storage, evm.context, nil, closure)
+
+	return evm
+}
+
```
Comment (d5c5ceb0) [inline sealEVM.go:96]: for test

File: storage/cache.go [hunk 2]
```diff
@@ -84,14 +84,15 @@ func MergeResultCache(src *ResultCache, to *ResultCache) {
 
 	for k, v := range src.Balance {
 		if to.Balance[k] != nil {
-			to.Balance[k].Balance.Add(v.Balance)
+			to.Balance[k] = v
```
Comment (AKACoder) [inline storage/cache.go]: same code in if & else, can we remove the if statement?

File: storage/storage.go [hunk 3]
```diff
@@ -76,17 +76,26 @@ func (s *Storage) SStore(n *evmInt256.Int, k *evmInt256.Int, v *evmInt256.Int) {
...
+	if err != nil {
+		return false
+	}
+
+	return balance.Cmp(amount.Int) >= 0
+}
+
 func (s *Storage) BalanceModify(address *evmInt256.Int, value *evmInt256.Int, neg bool) {
 	kString := address.AsStringKey()
+	s.Balance(address)
```
Comment (AKACoder) [inline storage/storage.go:90]: why call s.Balance here?

Comment (d5c5ceb0) [reply storage/storage.go:90]: just for reflashing cache.

**Pull request merged**